cdap-validation/bin/testing.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/python
+
+import test_helpers as helpers
+import json
+import ambari
+import cm
+
+##### TEST FRAMEWORK FUNCTIONS #####
+
+input_json = "input.json"
+
+# calls all other functions
+def testing(base_vars):
+    print 'start testing'
+
+    # prep input variables
+    cluster_vars = {}
+    input = read_input(input_json)
+    cluster_vars['host'] = input['params']['uri']
+    cluster_vars['username'],cluster_vars['password']= input['params']['userpass'].split(':')
+    cluster_vars['cluster'] = input['params']['cluster_name']
+    cluster_vars['modules'] = input['params']['modules']
+
+    # check for verbosity not set
+    if not input['params']['verbose']:
+        cluster_vars['verbose'] = 'default'
+    else:
+        cluster_vars['verbose'] = input['params']['verbose']
+
+    # find_modules
+    # hold on for now
+
+    # detect_installed_services
+    # hold on for now
+
+    # prep the url
+    find_api='/api'
+    if cluster_vars['host'].find(find_api) == -1:
+        cluster_vars['host'] += 'api/'
+    print 'uri=%s' % (cluster_vars['host'])
+
+    # validate_connection
+    print 'Validating connection'
+    api_tests = [{'cm': base_vars['cm']['api_test']},{'ambari': base_vars['ambari']['api_test']}]
+    version_test = validate_connection(base_vars, cluster_vars, api_tests)
+    print ''
+    print 'version_test=%s' % (version_test)
+
+    # now we determine the install manager from the info we have
+    print 'Determine Hadoop install manager'
+    install_manager = get_install_manager(version_test)
+    cluster_vars['manager'] = install_manager
+    print 'Hadoop install manager is %s' % (install_manager)
+
+    # get version for API call
+    if install_manager == 'cm':
+        cluster_vars['version'] = version_test
+    else:
+        cluster_vars['version'] = 'v1'
+    
+    cluster_vars['base_url'] = cluster_vars['host'] + cluster_vars['version']
+    
+    # get preliminary info
+    cluster_vars = get_install_manager_info(base_vars, cluster_vars, install_manager)
+      # do we need these (below) or can we just work with the dict values?
+    for key, value in cluster_vars.iteritems():
+        print "%s=%s" % (key, value)
+
+    # get and run commands (API commands for now)
+    get_commands(base_vars, cluster_vars)
+
+    #########################################
+
+    ##### MODULES #####
+    
+    # identify and run modules
+
+
+###########################################################################################
+# reads JSON object passed to the framework
+def read_input(input_file):
+    with open(input_file) as json_file:
+        data = json.load(json_file)
+    return data
+
+###def find_modules
+# navigate through ./module/* directories and look for module.json files
+# parse through those to determine modules that will be run
+# save (follow design)
+
+###def detect_installed_services
+# coming soon
+
+def validate_connection(base_info, cluster_info, api_tests):
```

Comment:
currently, this method adds no value to the ``test_api_connection`` method it wraps